plugin/procfs/procfs.cc

```diff
@@ -0,0 +1,514 @@
+                        eq_func->arguments()[0]->full_name(),
+                        field_name.c_str()) == 0) {
+        char buff[1024];
+        String *res;
+        String filter(buff, sizeof(buff), system_charset_info);
+        if (eq_func->arguments()[1] != nullptr &&
+            (res = eq_func->arguments()[1]->val_str(&filter))) {
+          eq_arg->append(res->c_ptr_safe(), res->length());
+          return false;
+        }
+      }
+    }
+  }
+  return true;
+}
+
+static bool get_like_condition_argument(Item *cond, std::string *like_arg,
+                                        const std::string &field_name) {
+  if (cond != 0 && cond->type() == Item::FUNC_ITEM) {
+    Item_func *func = static_cast<Item_func *>(cond);
+    if (func != nullptr && func->functype() == Item_func::LIKE_FUNC) {
+      Item_func_like *like_func = static_cast<Item_func_like *>(func);
+      if (like_func->arguments()[0]->type() == Item::FIELD_ITEM &&
+          my_strcasecmp(system_charset_info,
+                        like_func->arguments()[0]->full_name(),
+                        field_name.c_str()) == 0) {
+        char buff[1024];
+        String *res;
+        String filter(buff, sizeof(buff), system_charset_info);
+        if (like_func->arguments()[1] != nullptr &&
+            (res = like_func->arguments()[1]->val_str(&filter))) {
+          like_arg->append(res->c_ptr_safe(), res->length());
+          return false;
+        }
+      }
+    }
+  }
+  return true;
+}
+
+static bool get_in_condition_argument(Item *cond,
+                                      std::map<std::string, bool> &in_args,
+                                      const std::string &field_name) {
+  if (cond != 0 && cond->type() == Item::FUNC_ITEM) {
+    Item_func *func = static_cast<Item_func *>(cond);
+    if (func != nullptr && func->functype() == Item_func::IN_FUNC) {
+      Item_func_in *in_func = static_cast<Item_func_in *>(func);
+      if (in_func->arguments()[0]->type() == Item::FIELD_ITEM &&
+          my_strcasecmp(system_charset_info,
+                        in_func->arguments()[0]->full_name(),
+                        field_name.c_str()) == 0) {
+        char buff[1024];
+        String *res;
+        String filter(buff, sizeof(buff), system_charset_info);
+        for (uint i = 1; i < in_func->arg_count; ++i) {
+          if (in_func->arguments()[i] != nullptr &&
+              (res = in_func->arguments()[i]->val_str(&filter)) &&
+              res->length() > 0) {
+            in_args[std::string(res->c_ptr_safe(), res->length())] = true;
+          }
+        }
+        return false;
+      }
+    }
+  }
+  return true;
+}
+
+static void limited_glob_files(const std::string &path,
+                               const std::string &pattern, int max_results,
+                               std::vector<std::string> &files_found) {
+  static const constexpr char SYS_NAME[] = "/sys/";
+  static const constexpr int SYS_NAME_LEN = sizeof(SYS_NAME);
```

Comment:
previously this was strlen(SYS_NAME), which returns 5 (without the trailing 0). Now it's sizeof, which includes the zero, resulting in a length of 6. Based on the use I think this is what we actually want, just mentioning it so you can verify it.

Reply:
yes, exactly and I did myself in fact when I made this change.  The same goes for the PROC_NAME which would be strlen = 6 and sizeof = 7.